trust-quorum/test-utils/src/state.rs

```diff
@@ -0,0 +1,891 @@
+                // Save the configuration for later
+                self.all_coordinated_configs
+                    .insert_unique(latest_persisted_config.clone())
+                    .expect("unique");
+            }
+        }
+    }
+
+    /// Check postcondition assertions after initial configuration
+    pub fn postcondition_initial_configuration(&mut self) {
+        let (coordinator, msg) = self.nexus.reconfigure_msg_for_latest_config();
+
+        // The coordinator should have received the `ReconfigureMsg` from Nexus
+        if !self.faults.crashed_nodes.contains(coordinator) {
+            let (node, ctx) = self
+                .sut
+                .nodes
+                .get_mut(coordinator)
+                .expect("coordinator exists");
+            let mut connected_members = 0;
+            // The coordinator should start preparing by sending a `PrepareMsg` to all
+            // connected nodes in the membership set.
+            for member in
+                msg.members.iter().filter(|&id| id != coordinator).cloned()
+            {
+                if self.faults.is_connected(coordinator.clone(), member.clone())
+                {
+                    connected_members += 1;
+                    let msg_found = ctx.envelopes().any(|envelope| {
+                        envelope.to == member
+                            && envelope.from == *coordinator
+                            && matches!(
+                                envelope.msg.kind,
+                                PeerMsgKind::Prepare { .. }
+                            )
+                    });
+                    assert!(msg_found);
+                }
+            }
+            assert_eq!(connected_members, ctx.envelopes().count());
+
+            // The coordinator should be in the prepare phase
+            let cs = node.get_coordinator_state().expect("is coordinating");
+            assert!(matches!(cs.op(), CoordinatorOperation::Prepare { .. }));
+
+            // The persistent state should have changed
+            assert!(ctx.persistent_state_change_check_and_reset());
+            assert!(ctx.persistent_state().has_prepared(msg.epoch));
+            assert!(ctx.persistent_state().latest_committed_epoch().is_none());
+        }
+    }
+
+    /// Put any outgoing coordinator messages from the latest configuration on the wire
+    pub fn send_envelopes_from_coordinator(&mut self) {
+        let coordinator = {
+            let (coordinator, _) =
+                self.nexus.reconfigure_msg_for_latest_config();
+            coordinator.clone()
+        };
+        self.send_envelopes_from(&coordinator);
+    }
+
+    pub fn send_envelopes_from(&mut self, id: &PlatformId) {
+        let (_, ctx) = self.sut.nodes.get_mut(id).expect("node exists");
+        for envelope in ctx.drain_envelopes() {
+            let msgs =
+                self.bootstrap_network.entry(envelope.to.clone()).or_default();
```

Comment:
This was all moved as well. GitHub select when scrolling does not seem to be working as I'd hoped.